geode-core/src/main/java/org/apache/geode/internal/cache/BucketAdvisor.java

```diff
@@ -282,8 +282,11 @@ public boolean deposePrimary() {
           if (this.getBucket() instanceof BucketRegionQueue) {
             BucketRegionQueue brq = (BucketRegionQueue) this.getBucket();
             brq.decQueueSize(brq.size());
+            brq.incSecondaryQueueSize(brq.size());
```

Comment:
Should this be in the else block?

Reply:
No, it must be this way. When a member turned from primary to secondary, it increased secondary queue size, and decreased primary queue size, together. 

I have a test case with redundency=1, however, I'm thinking to add a dunit test with redundency=0, which requires some tricky code.